astropy/stats/bayesian_blocks.py

```diff
@@ -104,14 +104,19 @@ def bayesian_blocks(t, x=None, sigma=None,
     --------
     Event data:
 
+    >>> import warnings
     >>> t = np.random.normal(size=100)
-    >>> edges = bayesian_blocks(t, fitness='events', p0=0.01)
+    >>> with warnings.catch_warnings():
+    ...     warnings.simplefilter('ignore')  # Ignore p0 warning
```

Comment:
I'm not sure it's good to include these warning catchings in docstring examples. 

(currently I can't think of now a good way to catch it for our doctests, if there is no such way, it may be a good pytest-doctestplus feature request.)

Reply:
On the contrary, I feel like explicitly catching the warning tells the user to expect one, then the user will not be so surprised down the road?

Also, this is a must for when we proceed with #7928 . With Dan gone, I don't know if `pytest-doctestplus` will ever get any new features in, so this is the only way for now.

That, or, asking `bayesian_blocks` to stop throwing so many warnings about `p0`, which is also unlikely to happen.

As for this PR, if people feel strongly about this, I can back it out for now. But eventually, we have to come back to this discussion. `¯\_(ツ)_/¯`

Reply:
I'm not that pessimistic about `doctestplus`, it's a very useful piece if infrastructure that we heavily rely on, so it should be maintained and stuff we need will need  to be added. Dan refactored it out from the core, but it was never a solo project.
But maybe I'm overly optimistic.

I would rather use the `Warns` section to list the expected warnings than catch warnings in the example, but let's see what the overall preference.